.github/workflows/check-main.yml

```diff
@@ -5,6 +5,9 @@ on:
     branches:
       - main
 
+env:
+  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
```

Comment:
Just found out about this over the weekend and we're using a new enough nightly build that I thought it was worth a try. Looks like it might be shaving a couple mins off the CI build times compared to other PRs but I haven't done any proper timing or benchmarking of it yet.

Reply:
Cool, but would you mind splitting this out to a separate PR we can evaluate independently?